src/py/flwr/client/middleware/secure_aggregation/secaggplus_middleware.py

```diff
@@ -417,18 +466,9 @@ def _collect_masked_input(
         state.sk1_share_dict[src] = sk1_share
 
     # Fit client
-    parameters_bytes = cast(List[bytes], named_values[KEY_PARAMETERS])
-    parameters = [bytes_to_ndarray(w) for w in parameters_bytes]
-    if isinstance(state.client, Client):
-        fit_res = state.client.fit(
-            FitIns(parameters=ndarrays_to_parameters(parameters), config={})
-        )
-        parameters_factor = fit_res.num_examples
-        parameters = parameters_to_ndarrays(fit_res.parameters)
-    elif isinstance(state.client, NumPyClient):
-        parameters, parameters_factor, _ = state.client.fit(parameters, {})
-    else:
-        log(ERROR, "Client %d: fit function is missing.", state.sid)
+    fit_res = fit()
+    parameters_factor = fit_res.num_examples
+    parameters = parameters_to_ndarrays(fit_res.parameters)
```

Comment:
Same as above - we mustn't use `FitRes` and `Parameters` here since these are considered legacy messages

Reply:
I agree that SA MW should not use legacy messages. I plan to rewrite this part after we improve the `RecordSet` APIs and add utility functions to convert `ParametersRecord` to numpy arrays. But for now, I think we still have to work with legacy messages a little bit, as explained above.